fastiuk/sonic-swss-common
# ASIC Firmware update control

This PR depends on https://github.com/fastiuk/sonic-buildimage/pull/5

#### Why I did it
* Add an ability to control ASIC FW update
  * Ability to enable/disable FW update
  * Ability to choose FW source: image/user-defined

#### How to verify it
Build SONiC
or
When SONiC was already built:
```
rm -f target/python-wheels/bullseye/sonic_yang_models-1.0-py3-none-any.whl
BLDENV=bullseye make -f Makefile.work target/python-wheels/bullseye/sonic_yang_models-1.0-py3-none-any.whl
rm -f target/sonic-mellanox.bin
make target/sonic-mellanox.bin
```
Login into the switch.
Disable FW auto-update:
```
redis-cli -n 4 HSET "FIRMWARE|asic" "auto_update" "disable"
sudo reboot
```
See the results:
![Screenshot 2022-07-03 at 02 20 43](https://user-images.githubusercontent.com/35844154/177020334-66b0f9e3-4548-46e3-994c-34b9f89f38be.png)
Enable auto-update, change source to user-defined and copy FW image in place:
```
redis-cli -n 4 HSET "FIRMWARE|asic" "auto_update" "enable"
redis-cli -n 4 HSET "FIRMWARE|asic" "default" "user"
sudo mkdir -p /host/mlnx/asic/
sudo cp /tmp/fw-SPC2-rel-29_2010_2270-EVB.mfa /host/mlnx/asic/fw.mfa
sudo reboot
```
![Screenshot 2022-07-03 at 03 42 37](https://user-images.githubusercontent.com/35844154/177020394-2d7ee205-0d29-416d-a7ae-5ab91691efc6.png)

#### Which release branch to backport (provide reason below if selected)




#### Description for the changelog
* Add an ability to control ASIC FW update

#### Link to config_db schema for YANG module changes
[sonic-firmware.yang](https://github.com/fastiuk/sonic-buildimage/tree/firmware-update-control/src/sonic-yang-models/yang-models/sonic-firmware.yang)

#### Superproject PR :
| PR title | State |
| ----------------- | ----------------- |
| [[Nvidia] Firmware update control](https://github.com/fastiuk/sonic-buildimage/pull/5) | ![GitHub issue/pull request detail](https://img.shields.io/github/pulls/detail/state/fastiuk/sonic-buildimage/5) |

#### A picture of a cute animal
![1-46](https://user-images.githubusercontent.com/35844154/169010699-e61cb22f-c9ae-4d12-8ef1-64e3607eef4c.jpeg)
